plugins/network-elements/nuage-vsp/src/com/cloud/agent/api/VspResourceCommand.java

```diff
@@ -79,4 +79,41 @@ public String getProxyUserDomainuuid() {
     public boolean executeInSequence() {
         return false;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        VspResourceCommand that = (VspResourceCommand) o;
+
+        if (_childResource != null ? !_childResource.equals(that._childResource) : that._childResource != null)
+            return false;
+        if (_entityDetails != null ? !_entityDetails.equals(that._entityDetails) : that._entityDetails != null)
+            return false;
+        if (_method != null ? !_method.equals(that._method) : that._method != null) return false;
+        if (_proxyUserDomainuuid != null ? !_proxyUserDomainuuid.equals(that._proxyUserDomainuuid) : that._proxyUserDomainuuid != null)
+            return false;
+        if (_proxyUserUuid != null ? !_proxyUserUuid.equals(that._proxyUserUuid) : that._proxyUserUuid != null)
+            return false;
+        if (_resource != null ? !_resource.equals(that._resource) : that._resource != null) return false;
+        if (_resourceFilter != null ? !_resourceFilter.equals(that._resourceFilter) : that._resourceFilter != null)
+            return false;
+        if (_resourceId != null ? !_resourceId.equals(that._resourceId) : that._resourceId != null) return false;
+
+        return true;
+    }
+
+    @Override
+    public int hashCode() {
```

Comment:
This ``hashCode`` implementation should include attributes from the ``Command`` super class.